vspace/src/main/java/edu/asu/diging/vspace/core/services/impl/ExternalLinkManager.java

```diff
@@ -98,7 +116,40 @@ protected void removeFromLinkList(ISpace space, IExternalLink link) {
 
     @Override
     protected void deleteLinkRepo(IExternalLink link) {
-        externalLinkRepo.delete((ExternalLink)link);
+        externalLinkRepo.delete((ExternalLink) link);
+    }
+
+    @Override
+    public IExternalLinkDisplay createLink(String title, String id, float positionX, float positionY, int rotation,
+            String linkedId, String linkLabel, DisplayType displayType, byte[] linkImage, String imageFilename,
+            ExternalLinkDisplayMode howToOpen)
+            throws SpaceDoesNotExistException, ImageCouldNotBeStoredException, SpaceDoesNotExistException {
+
+        /*
+         * When createLink is called then inside updateLinkAndDisplay(link, displayLink)
+         * is called to save Link and Display link in the database
```

Comment:
that might be so but it doesn't explain why line 134 would be persisted. I believe Hibernate does that automatically but it should be commented on.